cognite/client/data_classes/_base.py

```diff
@@ -140,30 +140,36 @@ def _load(
         raise TypeError(f"Resource must be json str or dict, not {type(resource)}")
 
     def to_pandas(
-        self, expand: Sequence[str] = ("metadata",), ignore: list[str] | None = None, camel_case: bool = False
+        self,
+        expand_metadata: bool = True,
+        metadata_prefix: str = "metadata.",
+        ignore: list[str] | None = None,
+        camel_case: bool = False,
+        convert_timestamps: bool = True,
     ) -> pandas.DataFrame:
         """Convert the instance into a pandas DataFrame.
 
         Args:
-            expand (Sequence[str]): List of row keys to expand, only works if the value is a Dict. Will expand metadata by default.
-            ignore (list[str] | None): List of row keys to not include when converting to a data frame.
-            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`)
+            expand_metadata (bool): Expand the metadata into separate rows.
+            metadata_prefix (str): Prefix to use for the metadata rows, if expanded.
+            ignore (list[str] | None): List of row keys to skip when converting to a data frame. Is applied before expansions.
+            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`). Does not affect custom data like metadata.
+            convert_timestamps (bool): Convert known attributes storing CDF timestamps (milliseconds since epoch) to datetime. Does not affect custom data like metadata.
```

Comment:
Could this be a sequence of strings? Then, have a default with the known attributes, such that the user could just also specify the metadata field they want converted?

Reply:
Very uncertain about this suggestion as it increases complexity.

Reply:
Unsure! Could be added later without breaking, so I'll leave it for now.